types/smartystreets-javascript-sdk/index.d.ts

```diff
@@ -0,0 +1,378 @@
+    interface Metadata {
+        buildingDefaultIndicator: any;
+        carrierRoute: string;
+        congressionalDistrict: string;
+        countyFips: string;
+        countyName: string;
+        elotSequence: string;
+        elotSort: string;
+        isEwsMatch: boolean;
+        latitude: number;
+        longitude: number;
+        obeysDst: boolean;
+        precision: string;
+        rdi: string;
+        recordType: string;
+        timeZone: string;
+        utcOffset: number;
+        zipType: string;
+    }
+    interface Componenet {
+        cityName: string;
+        defaultCityName: string;
+        deliveryPoint: string;
+        deliveryPointCheckDigit: string;
+        extraSecondaryDesignator: any;
+        extraSecondaryNumber: any;
+        plus4Code: string;
+        pmbDesignator: any;
+        pmbNumber: any;
+        primaryNumber: string;
+        secondaryDesignator: string;
+        secondaryNumber: string;
+        state: string;
+        streetName: string;
+        streetPostdirection: string;
+        streetPredirection: string;
+        streetSuffix: string;
+        urbanization: string;
+        zipCode: string;
+    }
+}
+
+export namespace usZipcode {
+    class Lookup {
+        constructor(city?: string, state?: string, zipCode?: string, inputId?: string);
+        city: string;
+        state: string;
+        zipCode: string;
+        inputId: string;
+        result: [any];
+    }
+
+    class Result {
+        constructor(responseData: any);
+        inputIndex: string;
+        status: string;
+        reason: string;
+        valid: boolean;
+        cities: [City];
+        zipcodes: [ZipCode];
+    }
+    interface City {
+        city: string;
+        stateAbbreviation: string;
+        state: string;
+        mailableCity: string;
+    }
+    interface ZipCode {
+        zipcode: string;
+        zipcodeType: string;
+        defaultCity: string;
+        countyFips: any;
+        countyName: string;
+        latitude: number;
+        longitude: number;
+        precision: string;
+        stateAbbreviation: string;
+        state: string;
+        alternateCounties: [County];
+    }
+    interface County {
+        countyFips: any;
+        countyName: string;
+        stateAbbreviation: string;
+        state: string;
+    }
+}
+export namespace usAutocomplete {
+    class Lookup {
+        constructor(prefix: string);
+        prefix: string;
+        maxSuggestions: number;
+        cityFilter: [any];
+        stateFilter: [any];
+        prefer: [any];
+        preferRation: any;
+        geolocate: any;
+        geolocatePrecision: any;
+    }
+
+    class Suggestion {
+        constructor(responseData: any);
+        text: string;
+        streetLine: string;
+        city: string;
+        state: string;
+    }
+}
+export namespace usAutocompletePro {
+    class Lookup {
+        constructor(search: string);
+        result: [any];
+        search: string;
+		selected: any;
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/smartystreets-javascript-sdk/index.d.ts;
  Line: 294;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.